packages/asset-uploader/src/delete.ts

```diff
@@ -0,0 +1,41 @@
+import { DeleteObjectCommand } from "@aws-sdk/client-s3";
+import { prisma, Asset } from "@webstudio-is/prisma-client";
+import { unlink } from "fs/promises";
+import path from "path";
+import { deleteAssetInDb } from "./db";
+import { getImageLocalDirectory } from "./helpers/get-image-local-path";
+import { s3EnvVariables } from "./schema";
+import { getS3Client } from "./targets/s3/client";
+
+export const deleteAsset = async ({
+  id,
+  name,
+  dirname,
+}: {
+  id: string;
+  dirname: string;
+  name: string;
+}): Promise<Asset> => {
+  const currentAsset = await prisma?.asset.findUnique({
+    where: { id },
+  });
+  if (currentAsset && currentAsset.name) {
+    if (currentAsset.location === "REMOTE") {
+      const s3Envs = s3EnvVariables.parse(process.env);
+      await getS3Client().send(
+        new DeleteObjectCommand({
+          Bucket: s3Envs.S3_BUCKET,
+          Key: name,
+        })
+      );
+
+      return await deleteAssetInDb(id);
+    } else {
+      const directory = await getImageLocalDirectory(dirname);
+      await unlink(path.join(directory, name));
+
+      return await deleteAssetInDb(id);
+    }
+  }
+  throw new Error("Asset does not exist");
```

Comment:
can we throw early? like early return, easier to skip code I don't need to read

Reply:
done!|